### PR TITLE
replace README with rewritten one (includes JSON support, request headers)

### PR DESCRIPTION
**Pull request by davewhiteland**
Rewritten README was lagging behind changes so this is the rewrite as a single
commit. (Includes changes that had also been reflected in commits to README
for `Scraped::JSON` and request header passing example).

This does *not* include a separate `example.rb` file for tests, and it drops the
illustration which had non-exemplar example code snippets in it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
 # Scraped
 
-Write declarative scrapers in Ruby
+Write declarative scrapers in Ruby.
+
+If you need to write a webscraper (especially one that hits a page which lists a load of other pages, and jumps into each of _those_ pages to pull out the same data) the `scraped` gem will help you write it quickly and clearly.
```
Comment (tmtmtmtm) [inline README.md]: Does that 'especially' go too far in that direction? I find `scraped` really useful even when it's only scraping a single page.

File: README.md [hunk 2]
```diff
@@ -20,14 +22,29 @@ Or install it yourself as:
 
 ## Usage
 
+Scraped currently has support for working with HTML and JSON documents.
```
Comment (tmtmtmtm) [inline README.md]: "PRs for other formats are welcome"?

File: README.md [hunk 3]
```diff
@@ -20,14 +22,29 @@ Or install it yourself as:
...
+
+To write a scraper, start by creating a subclass of `Scraped::HTML` or `Scraped::JSON` for each _type_ of page you’re scraping. Then specify the (data) fields you want to extract from that. You can control the strategy used to get the page, and decorate the response you get back to make it easier to parse.
```
Comment (tmtmtmtm) [inline README.md]: The emphasis on 'type' makes it seem important, so I think we need something to spell out a bit more what that means (whether inplace here, or a pointer to the section that goes into that)

File: README.md [hunk 4]
```diff
@@ -37,30 +54,43 @@ class ExamplePage < Scraped::HTML
...
-JSON documents are handled in a similar way. Create a subclass of `Scraped::JSON` for each type of JSON document you wish to scrape.
+JSON documents are handled in a similar way. 
+
+If `http://example.com/data.json` returns something like this:
+
+```
+{
+  name: 'John Doe',
+  email: 'john_doe@example.com'
+}
```
Comment (tmtmtmtm) [inline README.md]: Probably better to use an example that's actually JSON…

**Review by chrismytton — APPROVED** [review]
Woo, Thanks @davewhiteland, this is excellent! 💖  LGTM! 🚀 

(And please forgive tardiness of my review! 😥 )

**Pull request merged**